zlib-rs/src/inflate.rs

```diff
@@ -863,7 +863,7 @@ impl<'a> State<'a> {
 
     fn lit(&mut self) -> ReturnCode {
         if self.writer.remaining() == 0 {
-            #[cfg(feature = "std")]
+            #[cfg(all(test, feature = "std"))]
```

Comment:
this and similar cases just limit what gets printed. in tests this could be useful information, in production you do not want this.

Reply:
Thanks, that is useful, apart form the crash, these outputs were messing up the structured output of my command line program. A library should never have prints like this, instead consider using the log or tracing frameworks if the outputs do belong in production.

Reply:
absolutely. It was unclear earlier to what extend these errors were fatal (and should be unreachable). We now know that they are part of the standard (de)compression workflow. So while in testing they may provide useful information, we don't want the prints in production for the reasons you mention.